rich/_log_render.py

```diff
@@ -8,29 +8,31 @@
     from .console import Console, ConsoleRenderable, RenderableType
     from .table import Table
 
+TimeFormatterCallable = Callable[[datetime], Text]
+
 
 class LogRender:
     def __init__(
         self,
         show_time: bool = True,
         show_level: bool = False,
         show_path: bool = True,
-        time_format: str = "[%x %X]",
+        time_format: Union[str, TimeFormatterCallable] = "[%x %X]",
         level_width: Optional[int] = 8,
     ) -> None:
         self.show_time = show_time
         self.show_level = show_level
         self.show_path = show_path
         self.time_format = time_format
         self.level_width = level_width
-        self._last_time: Optional[str] = None
+        self._last_time: Optional[Text] = None
 
     def __call__(
         self,
         console: "Console",
         renderables: Iterable["ConsoleRenderable"],
         log_time: datetime = None,
-        time_format: str = None,
+        time_format: Union[str, TimeFormatterCallable] = None,
```

Comment:
I changed the type but since `time_format` is coming directly from the `logging` package's `Formatter` class, and is typed as `datefmt: Optional[str]` then we'll never actually get a callable?

Reply:
It can also come from the Console.log method, which also uses LogRender